daemon/top_unix_test.go

```diff
@@ -64,6 +90,29 @@ func TestContainerTopParsePSOutput(t *testing.T) {
    43 bar
   100 baz
 `), []uint32{42, 43}, true},
+		{[]byte(`  PID COMMAND
+   42root
+`), []uint32{42}, false},
+		{[]byte(`  PID COMMAND
+   42root
+   43 bar
+   100 baz
+`), []uint32{42, 43, 100}, false},
+		{[]byte(`  PID COMMAND
+   43 bar
+   42root
+   100 baz
+`), []uint32{43, 42, 100}, false},
+		{[]byte(`  PID COMMAND
+   43 bar
+   100 baz
+   42root
+`), []uint32{43, 100, 42}, false},
+		{[]byte(`  PID COMMAND
+   43 bar
+   100123
+   42root
+`), []uint32{43, 100123, 42}, false},
 	}
```

Comment:
What happens if a command that starts with a numeric character is running?

Reply:
My change intentionally does nothing new in this situation, the command would behave the same as it does today.

That being said, we should probably error out in this case since the PID value was be incorrect but it looks like the command would complete successfully.

The challenge is detecting this condition.  On the surface it seems like you should be able to make sure the number of headers matches the number of fields in the line.  But some header values can contain fields with spaces, so you would have to conditionally ignore these spaces.

Another option could be to get a list of all PIDs and compare the parsed PID value with the list to make sure it’s a valid value.  I wasn’t able to find a ‘ps’ command option that worked universally across all linux hosts.  This would definitely be the best option since the PID value could be reliably found.